WordPress/gutenberg
# Editor: Deprecate `layout` attribute

Related: #7234
Related: #7841

This pull request seeks to deprecate `layout`-related behaviors. As of #7234, no core functionality leverages this feature. Its need has been superseded by the additional functionality of block insertion restrictions via `allowedBlocks` and `parent`. It's continued existence has already proven to be a maintenance burden, the removal of which (not done here) will enable a significant simplification to the editor implementation.

**Testing instructions:**

There should be no deprecated warnings when using the editor, since no functionality exists leveraging `layouts`.

Verify there are no regressions (nor warnings) in the behavior of drag-and-drop.

Verify that an `InnerBlocks` implementation using `layouts` still works, but logs warnings (needs example).
